### ISSUE TITLE
[Feature]: Use .fetch() instead of .resolve() for simple command args

### ISSUE THREAD

**Issue by lolPants** [oceanroleplay/discord.ts#528]
### Feature?

When using a simple command arg for a channel or user, the library internally uses `.resolve()` which will attempt to pull the target from the client's cache. This means if the target is not cached, `null` will be returned.

Unfortunately moving to use `.fetch()` would require async and some error handling logic. The other option would be to explicitly document the behaviour and ensure that where appropriate users handle the `null` case.

### Package

discordx

**Comment by oceanroleplay** [comment]
@lolPants have you encountered this issue?

**Comment by VictoriqueMoe** [comment]
Agree, should check the cache.
https://discord.js.org/#/docs/discord.js/stable/typedef/BaseFetchOptions

The async handler and the exception is not an issue, any issue that happens during the internal resolution of DJS due to a propagation should be thrown back to the user. 

You should never get null

**Comment by lolPants** [comment]
@oceanroleplay yeah, was looking to port an old Discord.js v12 codebase to something more modern and ran into this issue pretty quickly.

**Comment by VictoriqueMoe** [comment]
> @oceanroleplay yeah, was looking to port an old codebase using Discord.js v12 to something more modern and ran into this issue pretty quickly.

just an FYI, `@SimpleCommand` is a last case for a command that can not be used for Slash commands.

**Comment by oceanroleplay** [comment]
> @oceanroleplay yeah, was looking to port an old codebase using Discord.js v12 to something more modern and ran into this issue pretty quickly.

sure thanks for reporting, This issue will be resolved very soon.

**Comment by oceanroleplay** [comment]
@lolPants I request you review https://github.com/oceanroleplay/discord.ts/pull/529

If you think there can be more improvements, let me know.

**Comment by oceanroleplay** [comment]
Install v9.3.0 for fix 🎉

**Comment by lolPants** [comment]
Was a little busy today so didn't get around to a review, but all looks good!

Thanks for working on this 😄
